lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/GeoPolygon.java

```diff
@@ -80,25 +79,19 @@ private GeoRectangle calcBoundingBox(List<GeoPoint> vertices) {
 
     @Override
     public boolean contains(GeoPoint point) {
-        final Ray rayFromPoint = new Ray();
-        point.toVector3d(rayFromPoint.origin);
-        rayFromPoint.origin.set(point.getLongitude(), 0, -point.getLatitude());
-        rayFromPoint.direction.set(1, 0, 0);
-
-        int intersections = 0;
-        final Vector3d edgeStart = new Vector3d();
-        final Vector3d edgeEnd = new Vector3d();
-        GeoPoint prev = null;
-        for (GeoPoint curr : getVertices()) {
-            if (prev != null) {
-                edgeStart.set(prev.getLongitude(), 0, -prev.getLatitude());
-                edgeEnd.set(curr.getLongitude(), 0, -curr.getLatitude());
-                boolean intersect = rayFromPoint.intersectsLineSegmentXZ(edgeStart, edgeEnd);
-                intersections += intersect ? 1 : 0;
-            }
-            prev = curr;
+        float[] verticesXValues = new float[vertices.size()];
+        float[] verticesYValues = new float[vertices.size()];
+
+        for (int i = 0; i < vertices.size(); i++) {
+            GeoPoint geoPoint = vertices.get(i);
+            verticesXValues[i] = (float) geoPoint.toCartesian().getX();
+            verticesYValues[i] = (float) geoPoint.toCartesian().getY();
```

Comment:
I don't wanna be picky, but we could half the number of conversions to cartesian by converting to CartesianPoint once and use their X/Y coordinates afterwards.

Reply:
You should be picky :+1: